ToggleButton/ToggleButton.js

```diff
@@ -0,0 +1,212 @@
+/**
+ * Agate styled button components and behaviors.
+ *
+ * @example
+ * <Button small>Hello Enact!</Button>
+ *
+ * @module agate/Button
+ * @exports Button
+ * @exports ButtonBase
+ * @exports ButtonDecorator
+ */
+
+import kind from '@enact/core/kind';
+import Pure from '@enact/ui/internal/Pure';
+import Toggleable from '@enact/ui/Toggleable';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import Button from '../Button';
+// import {MarqueeDecorator} from '../Marquee';
+import Skinnable from '../Skinnable';
+
+import css from './ToggleButton.less';
+
+
+/**
+ * A stateless [Button]{@link agate/Button.Button} that can be toggled by changing its
+ * `selected` property.
+ *
+ * @class ToggleButtonBase
+ * @memberof agate/ToggleButton
+ * @extends agate/Button.Button
+ * @ui
+ * @public
+ */
+const ToggleButtonBase = kind({
+	name: 'ToggleButton',
+
+	propTypes: /** @lends agate/Button.ButtonBase.prototype */ {
+		/**
+		 * The background-color opacity of this button.
+		 *
+		 * * Values: `'translucent'`, `'lightTranslucent'`, `'transparent'`
+		 *
+		 * @type {String}
+		 * @public
+		 */
+		backgroundOpacity: PropTypes.oneOf(['translucent', 'lightTranslucent', 'transparent']),
+
+		/**
+		 * The string to be displayed as the main content of the toggle button.
+		 *
+		 * If `toggleOffLabel` and/or `toggleOnLabel` are provided, they will be used for the
+		 * respective states.
+		 *
+		 * @type {Node}
+		 * @public
+		 */
+		children: PropTypes.node,
+
+		/**
+		 * Disables the button.
+		 *
+		 * @type {Boolean}
+		 * @default false
+		 * @public
+		 */
+		disabled: PropTypes.bool,
+
+		/**
+		 * Enforces a minimum width on the Button.
+		 *
+		 * *NOTE*: This property's default is `true` and must be explicitly set to `false` to allow
+		 * the button to shrink to fit its contents.
+		 *
+		 * @type {Boolean}
+		 * @default true
+		 * @public
+		 */
+		minWidth: PropTypes.bool,
```

Comment:
I notice that when we toggle between labels of varying length ("off" and "on"), the width of the `ToggleButton` will grow/shrink somewhat, which seems unexpected when using `minWidth`. Would this prop stabilize that behavior? If not, we may want to look into adjusting that specifically.

Reply:
I'll remove this for now, and have leave the min-width up to the consumer